spec/lib/guard/puma_spec.rb

```diff
@@ -61,40 +73,125 @@
         guard.start
       end
     end
+
+    describe "UI message" do
+      before do
+        allow(guard.runner).to receive(:start)
+      end
+
+      context "no :config in options" do
+        it "contains port" do
+          expect(Guard::UI).to receive(:info).with(
+            /^Guard::Puma will now start your app on port 4000 using \w+ environment.$/
```

Comment:
Let's just assert on the `/ on port 4000/` instead of the whole message.

Reply:
I've add the whole message because of testing not only text about port (it's a different between two contexts), but whole message, as in spec titles (`Guard::Puma#start UI message`).

Otherwise we don't test the whole message somewhere, and message like `"on port 4000"` or `"start on port 4000"` can pass 1 or 2 specs, but it's incorrect message. Even with 5 specs for 1 message we will not sure in order of these parts from separate specs.

So… are you sure about this short Regexp? Specs perfectly show the current behavior now, and the message can be divided (and simplified) with more specs incoming, I think.

Reply:
That's true, good point. However, I really only like to test key words, like `restarting` and then anything that can change, like the `port` message or the environment.

Said another way: I wouldn't want this spec to fail if I changed the message to say `Guard::Puma will now start on port 4000 using development environment` by removing `your app`.

IMO the message really needs to include action, port and environment, and anything else isn't worth testing.

Reply:
> Said another way: I wouldn't want this spec to fail if I changed the message to say Guard::Puma will now start on port 4000 using development environment by removing your app.

OK, here is example:

Why `restarting`, `restarted` and `not restarted` have `Puma %do something%`, but `start` has `Guard::Puma will now start`? I think, it's not good.

And this is noticable from specs.

And if you want to change, by TDD you should change specs at first → they should start to fail → you should change code → specs should pass again.

No? Anyway short Regexps?

(And can I change `start` UI message to `Puma starting on port N in M environment.`?)

Reply:
They probably don't have that out of laziness from when I ported this originally 😉 
I'd be happy to see all that cleaned up, and it could go in this PR if you like.

Reply:
Done, as you want.

Also `starting` message bringing to the form of `restarting` and `restarted` messages.